session/txn.go

```diff
@@ -53,6 +53,17 @@ type TxnState struct {
 	doNotCommit error
 }
 
+var _ kv.SafeStore = &TxnState{}
+
+// SetContract implements the kv.SafeStore interface.
+func (st *TxnState) SetContract(key kv.Key, contract kv.ContractType) {
+	if st.Transaction != nil {
+		if raw, ok := st.Transaction.(kv.SafeStore); ok {
```

Comment:
I think `SetContract` needs to return an error if the store does not support it.

Reply:
`SetContract` is optional, if a store does not support it, it's safe to ignore contract check.